### ISSUE TITLE
Temperature ranges of Mini Color (pid 49) BR30 (pid 63) differ with those in LIFX App

### ISSUE THREAD

**Issue by wborn** [LIFX/products#12]
Hi @delfick!

In our openHAB community a user [reports](https://community.openhab.org/t/lifx-binding-temperature-dimmer-doesnt-cover-full-colour-temperature-range/115330?u=wborn) that the temperature ranges of the [Mini Color (pid 49)](https://github.com/LIFX/products/blob/2adf13f024ad0ff663a3c10f2a0704cc9e472f33/products.json#L376-L390) and [BR30 (pid 63)](https://github.com/LIFX/products/blob/2adf13f024ad0ff663a3c10f2a0704cc9e472f33/products.json#L541-L555) are 1500K-9000K in the LIFX App whereas the products.json lists these as 2500K-9000K.

Can you check if this is an issue in the products.json? Maybe there are similar issues?

**Comment by delfick** [comment]
Hello!

This one is a bit tricky because the range changed with newer firmware, which means I'd have to change the format of the product.json to support the ability to specify when it changed or put in a value that's wrong for older firmware.

**Comment by Shakesbeard** [comment]
hmm maybe there is no need to change anything. 

Why not add an additional property. Like an array or object indicating the latest firmware version for the contained range pair. And keep the most recent value pair as "the value" like they are now.
Would enable app developers able to support older firmwares if they are up to and you can see that there was something changed within the project.json. Also the format would be fully backwards compatible.

**Comment by delfick** [comment]
I could do that. I'm hesitant to make the structure messier (I already hate how we did it for extended firmware). I'm back at work later in the week, I'll set myself a reminder to make some kind of change.

In the meantime, do you all have suggestions on a specific format that would easy to work with? (I have my own ideas, but will be good to get other opinions too)

**Comment by wborn** [comment]
Thanks for your quick explanation! We currently don't use the JSON directly neither do we support multiple firmware versions. I'd probably base the values we use on that of the most recent firmware. Maybe later I find some time to properly support multiple firmware versions. :slightly_smiling_face:

**Comment by delfick** [comment]
Oh yeah, fair enough. Photons is similar in that it has its [own product registry](https://github.com/delfick/photons/blob/main/modules/photons_products/registry.py#L198) which isn't generated from the json. And has the same problem where it assumes kelvin values are constant.

**Comment by delfick** [comment]
Ok, so it took me a while, but I managed to find some time for this. And with the help of the firmware team I was able to check all the kelvin values we have in our public registry here. Turns out the two pids in this particular issue have actually supported 1500-9000 since 3.42 which was from 2018. So those two can be assumed as 1500-9000.

For the others, I've introduced a structure for specifying when the kelvin ranges change. I encourage comments and opinions on this PR - https://github.com/LIFX/products/pull/13 :)

**Comment by wborn** [comment]
Thanks for adding those updates @delfick! :+1: Looks all very usable and it will be easy to manage any firmware feature upgrade this way. :slightly_smiling_face:  I might want to add the info in a similar way to my own code to support multiple firmware versions.

**Comment by delfick** [comment]
Thanks :)

I also want to make Photons registry understand kelvin can change with firmware version. When I find the time.

**Comment by ananda-brilliant** [comment]
@delfick I realize an issue reporter isn't the best place to bring this up, but is there some reason why the bulbs themselves just can't advertise their capabilities in response to some API call? It seems much simpler than having to maintain a hard-coded list like this - is there a technical reason?

**Comment by delfick** [comment]
it's not in the protocol and hasn't been prioritised to be added. Personally I want that, but it's unlikely to happen anytime soon.

**Comment by delfick** [comment]
I've merged that PR, so the product.json should now have correct kelvin ranges :) (until the next time someone points out one of them is wrong...)

**Comment by wborn** [comment]
I found some time to support the different firmware versions based on the products.json in openHAB with https://github.com/openhab/openhab-addons/pull/10384. :slightly_smiling_face: 

After having a closer look at the firmware upgrade definitions I did notice that [PID 39](https://github.com/LIFX/products/blob/f90991802018b6f255250d65e38870154a6f8297/products.json#L418-L444) has a 2.80 firmware upgrade with the same temperature range as the initial firmware (1500-9000K). Probably the initial firmware should have a 2500-9000K temperature range?
